test/extended/util/client.go

```diff
@@ -563,6 +577,14 @@ func (c *CLI) RESTMapper() meta.RESTMapper {
 	return ret
 }
 
+func (c *CLI) setupStaticConfigsFromManifests() {
+	if len(c.staticConfigManifestDir) > 0 {
+		err, objects := collectConfigManifestsFromDir(c.staticConfigManifestDir)
+		o.Expect(err).ToNot(o.HaveOccurred())
```

Comment:
There is a bug here.

We start with a normal dry run execution to get the test list.
```bash
$ ./openshift-tests run openshift/conformance --provider none -v 2 --dry-run  | wc -l
Unable to get server version through discovery client, skipping apigroup check in the dry-run mode: the server has asked for the client to provide credentials
3437
```
Then we export the variable
```bash
$ export STATIC_CONFIG_MANIFEST_DIR=$(pwd)/tmp
$ ls -l $STATIC_CONFIG_MANIFEST_DIR 
total 8
-rw-rw-r--. 1 pacevedo pacevedo 310 Mar 21 23:10 infra.yaml
-rw-rw-r--. 1 pacevedo pacevedo 148 Mar 21 23:10 net.yaml
$ ./openshift-tests run openshift/conformance --provider none -v 2 --dry-run  | wc -l
Unable to get server version through discovery client, skipping apigroup check in the dry-run mode: the server has asked for the client to provide credentials
213
```
There is a huge difference in the amount of tests that will be executed, and it only happens after exporting the variable, all other things equal.
The error comes from this specific line. The Expect call will check if gomega was [configured](https://github.com/openshift/origin/blob/master/vendor/github.com/onsi/gomega/gomega_dsl.go#L221).
Which checks the [configuration for the default gomega instance](https://github.com/openshift/origin/blob/master/vendor/github.com/onsi/gomega/gomega_dsl.go#L168), which is what we are using.
Which checks the [presence of special handlers](https://github.com/openshift/origin/blob/master/vendor/github.com/onsi/gomega/internal/gomega.go#L26), which are [not configured](https://github.com/openshift/origin/blob/master/vendor/github.com/onsi/gomega/internal/gomega.go#L18-L22) in the [default instance](https://github.com/openshift/origin/blob/master/vendor/github.com/onsi/gomega/gomega_dsl.go#L47).

The `run-test` command does [configure a fail handler](https://github.com/openshift/origin/blob/master/cmd/openshift-tests/openshift-tests.go#L488), for example.

Reply:
My bad. I was checking error with gingko.Expect before it was properly initialized.